src/cmd/troubleshoot/image_helper.go

```diff
@@ -0,0 +1,125 @@
+package troubleshoot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	removeSchemaRegex      = regexp.MustCompile("^.*//(.*)$")
+	removeApiEndpointRegex = regexp.MustCompile("^(.*)/[^/]*$")
+	registryRegex          = regexp.MustCompile(`^(.*)/linux.*$`)
+	imageRegex             = regexp.MustCompile(`^.*/(linux.*)$`)
+)
+
+type imageInfo struct {
+	registry string
+	image    string
+	version  string
+}
+
+func splitImageName(imageName string) (imageInfo, error) {
+	// some image path examples that work with this function
+	//   aaa00000.dynatrace.com/linux/oneagent
+	//   aaa00000.dynatrace.com/linux/activegate:1.123
+
+	imgInfo := imageInfo{}
+
+	registryMatches := registryRegex.FindStringSubmatch(imageName)
+	if len(registryMatches) < 2 {
+		return imageInfo{}, fmt.Errorf("invalid image - registry not found (%s)", imageName)
+	}
+	imgInfo.registry = registryRegex.FindStringSubmatch(imageName)[1]
+
+	imageMatches := imageRegex.FindStringSubmatch(imageName)
+	if len(imageMatches) < 2 {
+		return imageInfo{}, fmt.Errorf("invalid image - endpoint not found (%s)", imageName)
+	}
+	imgInfo.image = imageRegex.FindStringSubmatch(imageName)[1]
+
+	imgInfo.version = ""
+
+	// check if image has version set
+	fields := strings.Split(imgInfo.image, ":")
+	if len(fields) == 1 || len(fields) >= 2 && fields[1] == "" {
+		// no version set, default to latest
+		imgInfo.version = "latest"
+		logInfof("using latest image version")
+	} else if len(fields) >= 2 {
+		imgInfo.image = fields[0]
+		imgInfo.version = fields[1]
+		logInfof("using custom image version: %s", imgInfo.version)
+	} else {
+		return imageInfo{}, fmt.Errorf("invalid version of the image {\"image\": \"%s\"}", imgInfo.image)
+	}
+	return imgInfo, nil
+}
+
+func splitCustomImageName(imageURL string) (imageInfo, error) {
+	imgInfo := imageInfo{}
+
+	customImageRegistryRegex := regexp.MustCompile(`^(.*?)/(.*)$`)
```

Comment:
nitpick: with regexes I think a comment is necessary to describe what they are doing. maybe with an example, so they don't have to be manually parsed when read